### PR TITLE
feat: support structured outputs for Anthropic models

### LINKED ISSUE #4174: [FEATURE] Anthropic: support structured outputs
Many of LC4j integrations already support structured outputs, see https://docs.langchain4j.dev/tutorials/structured-outputs/#json-schema and "JSON Schema" column on https://docs.langchain4j.dev/integrations/language-models/

Anthropic has recently introduced structured outputs, see https://platform.claude.com/docs/en/build-with-claude/structured-outputs

## Implementation:
See how this feature is implemented in other proviuders, such as OpenAI and Ollama.

## Testing:
Remove `supportsJsonResponseFormat*()` methods from `AnthropicChatModelIT` and `AnthropicStreamingChatModelIT`, make sure all tests are green

### PR DESCRIPTION
**Pull request by diegoberriosr**
## Issue

Closes #4174 

## Change

Added structured output support for Anthropic models.

- Introduced `responseFormat()` and `supportedCapabilities()` methods for `AnthropicChatModel` and `AnthropicStreamingChatModel`.
- Modeled JSON schemas in Anthropic requests via `AnthropicResponseFormat`.
- Added a helper method in `AnthropicMapper` (`toAnthropicMap()`) to map to the JSON schema format expected by the Anthropic API. Some features (such as unions and complex types in enums) are still not supported, so using the traditional `JsonSchemaUtils.toMap()` does not work for certain edge cases.
- Removed `supportsJsonResponseFormat*()` methods from `AnthropicChatModelIT` and `AnthropicStreamingChatModelIT`, and ensured all tests pass, as requested in the issue description. Some tests had to be overridden, since structured outputs are still a beta feature in Anthropic so models require a special configuration.

## General checklist

- [X] There are no breaking changes (API, behaviour)
- [X] I have added unit and/or integration tests for my change
- [X] The tests cover both positive and negative cases
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [X] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

### CODE REVIEW

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicChatModel.java [hunk 1]
```diff
@@ -415,6 +428,16 @@ public AnthropicChatModelBuilder customParameters(Map<String, Object> customPara
...
         }
 
+        public AnthropicChatModelBuilder supportedCapabilities(Capability... supportedCapabilities) {
+            this.supportedCapabilities = Arrays.stream(supportedCapabilities).collect(Collectors.toSet());
+            return this;
+        }
+
+        public AnthropicChatModelBuilder supportedCapabilities(Set<Capability> supportedCapabilities) {
+            this.supportedCapabilities = Set.copyOf(supportedCapabilities);
+            return this;
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicChatModel.java:446]: Nit: no need to copy here, it will be copied in ctor

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicStreamingChatModel.java [hunk 2]
```diff
@@ -396,6 +409,16 @@ public AnthropicStreamingChatModelBuilder customParameters(Map<String, Object> c
             return this;
         }
 
+        public AnthropicStreamingChatModelBuilder supportedCapabilities(Capability... supportedCapabilities) {
+            this.supportedCapabilities = Arrays.stream(supportedCapabilities).collect(Collectors.toSet());
+            return this;
+        }
+
+        public AnthropicStreamingChatModelBuilder supportedCapabilities(Set<Capability> supportedCapabilities) {
+            this.supportedCapabilities = Set.copyOf(supportedCapabilities);
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicStreamingChatModel.java]: Same here

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java [hunk 3]
```diff
@@ -26,6 +26,7 @@ public class AnthropicCreateMessageRequest {
     public Integer topK;
     public List<AnthropicTool> tools;
     public AnthropicToolChoice toolChoice;
+    public AnthropicResponseFormat outputFormat;
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java]:
```suggestion
    public AnthropicOutputFormat outputFormat;
```
For consistency

**Comment by dliubarskyi** [comment]
> Some features (such as unions and complex types in enums) are still not supported, so using the traditional JsonSchemaUtils.toMap() does not work for certain edge cases.

Could you please elaborate? Can't we reuse existing toMap() functionality and ignore unsupported features of Anthropic?

**Review by dliubarskyi — COMMENTED** [review]
@diegoberriosr well done, thank you! Please update the documentation page https://docs.langchain4j.dev/integrations/language-models/ (add a green tick for Anthropic structured outputs)

**Comment by diegoberriosr** [comment]
> > Some features (such as unions and complex types in enums) are still not supported, so using the traditional JsonSchemaUtils.toMap() does not work for certain edge cases.
> 
> Could you please elaborate? Can't we reuse existing toMap() functionality (...)?

@dliubarskyi Yes, I was referring to Anthropic requiring `additionalProperties=false` for object schemas, while simultaneously not supporting complex types and unions in enum schemas.

Setting strict to true in the existing `toMap()` will append `additionalProperties=false` to the final schema, but if the object schema contains enums, it will also generate enum schemas with unions unless the field is explicitly marked as required. This causes `AbstractBaseChatModelIT`’s `should_execute_a_tool_then_answer_respecting_JSON_response_format_with_schema` test to fail:

```java
dev.langchain4j.exception.InvalidRequestException: {"type":"error","error":{"type":"invalid_request_error","message":"output_format.schema: Invalid schema: Enum value 'sunny' does not match declared type '['string', 'null']'"},"request_id":"(...)"}
```

Because of this, I thought perhaps it would be better to handle cases involving object schemas (object and array types) separately, and then let toMap() handle the rest.

**Comment by diegoberriosr** [comment]
> (...) and ignore unsupported features of Anthropic?

In this case, by “ignoring unsupported features,” do you mean using the existing `toMap()` and overriding the failing test (for example, disabling it or adapting it so that enums are marked as required in the schema)?

Thank you!

**Comment by diegoberriosr** [comment]
> @diegoberriosr well done, thank you! Please update the documentation page https://docs.langchain4j.dev/integrations/language-models/ (add a green tick for Anthropic structured outputs)

@dliubarskyi Thanks a lot! 

Done :))

**Comment by dliubarskyi** [comment]
Could you please apply this patch? I can't push in this branch

```
Subject: [PATCH] WIP
---
Index: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java
--- a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -71,7 +71,7 @@
                 .topP(chatRequest.topP())
                 .topK(chatRequest.topK())
                 .thinking(thinking)
-                .outputFormat(toAnthropicResponseFormat(chatRequest.responseFormat()))
+                .outputFormat(toAnthropicOutputFormat(chatRequest.responseFormat()))
                 .customParameters(customParameters);
 
         List<AnthropicTool> tools = new ArrayList<>();
@@ -97,7 +97,7 @@
         return requestBuilder.build();
     }
 
-    public static AnthropicOutputFormat toAnthropicResponseFormat(ResponseFormat responseFormat) {
+    public static AnthropicOutputFormat toAnthropicOutputFormat(ResponseFormat responseFormat) {
         if (responseFormat == null || responseFormat.type() == TEXT || responseFormat.jsonSchema() == null) {
             return null;
         }
Index: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java
--- a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicCreateMessageRequest.java	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -219,6 +219,7 @@
                         .topK(this.topK)
                         .tools(this.tools)
                         .toolChoice(this.toolChoice)
+                        .outputFormat(this.outputFormat)
                         .thinking(this.thinking)
                         .metadata(this.metadata)
                         .customParameters(this.customParameters);
Index: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java
--- a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -1,8 +1,8 @@
 package dev.langchain4j.model.anthropic.internal.api;
 
 import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
-import static dev.langchain4j.model.anthropic.internal.api.AnthropicResponseFormatType.JSON_SCHEMA;
-import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.toAnthropicMap;
+import static dev.langchain4j.model.anthropic.internal.api.AnthropicOutputFormatType.JSON_SCHEMA;
+import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.toAnthropicSchema;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
@@ -22,7 +22,7 @@
 public class AnthropicOutputFormat {
 
     @JsonProperty
-    private final AnthropicResponseFormatType type;
+    private final AnthropicOutputFormatType type;
 
     @JsonProperty
     private final Map<String, Object> schema;
@@ -32,7 +32,7 @@
         this.schema = builder.schema;
     }
 
-    public AnthropicResponseFormatType getType() {
+    public AnthropicOutputFormatType getType() {
         return type;
     }
 
@@ -43,7 +43,7 @@
     public static AnthropicOutputFormat fromJsonSchema(JsonSchema schema) {
         return AnthropicOutputFormat.builder()
                 .type(JSON_SCHEMA)
-                .schema(toAnthropicMap(schema.rootElement()))
+                .schema(toAnthropicSchema(schema.rootElement()))
                 .build();
     }
 
@@ -74,10 +74,10 @@
     @JsonIgnoreProperties(ignoreUnknown = true)
     @JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
     public static class Builder {
-        private AnthropicResponseFormatType type;
+        private AnthropicOutputFormatType type;
         private Map<String, Object> schema;
 
-        public Builder type(AnthropicResponseFormatType type) {
+        public Builder type(AnthropicOutputFormatType type) {
             this.type = type;
             return this;
         }
Index: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicResponseFormatType.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicResponseFormatType.java b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormatType.java
rename from langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicResponseFormatType.java
rename to langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormatType.java
--- a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicResponseFormatType.java	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormatType.java	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -2,7 +2,8 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public enum AnthropicResponseFormatType {
+public enum AnthropicOutputFormatType {
+
     @JsonProperty("json_schema")
     JSON_SCHEMA,
 }
Index: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java
--- a/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -385,11 +385,11 @@
                 .build();
     }
 
-    public static Map<String, Object> toAnthropicMap(JsonSchemaElement schemaElement) {
-        return toAnthropicMap(schemaElement, true);
+    public static Map<String, Object> toAnthropicSchema(JsonSchemaElement schemaElement) {
+        return toAnthropicSchema(schemaElement, true);
     }
 
-    private static Map<String, Object> toAnthropicMap(JsonSchemaElement schemaElement, boolean required) {
+    private static Map<String, Object> toAnthropicSchema(JsonSchemaElement schemaElement, boolean required) {
         if (schemaElement instanceof JsonObjectSchema objectSchema) {
             Map<String, Object> map = new LinkedHashMap<>();
 
@@ -404,7 +404,7 @@
                     .properties()
                     .forEach((property, value) -> properties.put(
                             property,
-                            toAnthropicMap(value, objectSchema.required().contains(property))));
+                            toAnthropicSchema(value, objectSchema.required().contains(property))));
             map.put("properties", properties);
 
             if (objectSchema.required() != null) {
@@ -431,7 +431,7 @@
             }
 
             if (arraySchema.items() != null) {
-                map.put("items", toAnthropicMap(arraySchema.items()));
+                map.put("items", toAnthropicSchema(arraySchema.items()));
             } else {
                 map.put("items", Collections.emptyMap());
             }
@@ -446,7 +446,7 @@
 
     private static Map<String, Map<String, Object>> mapDefs(Map<String, JsonSchemaElement> defs) {
         Map<String, Map<String, Object>> map = new LinkedHashMap<>();
-        defs.forEach((property, schema) -> map.put(property, toAnthropicMap(schema)));
+        defs.forEach((property, schema) -> map.put(property, toAnthropicSchema(schema)));
 
         return map;
     }
Index: langchain4j-anthropic/src/main/resources/META-INF/native-image/dev.langchain4j/langchain4j-anthropic/reflect-config.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/main/resources/META-INF/native-image/dev.langchain4j/langchain4j-anthropic/reflect-config.json b/langchain4j-anthropic/src/main/resources/META-INF/native-image/dev.langchain4j/langchain4j-anthropic/reflect-config.json
--- a/langchain4j-anthropic/src/main/resources/META-INF/native-image/dev.langchain4j/langchain4j-anthropic/reflect-config.json	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/main/resources/META-INF/native-image/dev.langchain4j/langchain4j-anthropic/reflect-config.json	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -72,6 +72,24 @@
         "allPublicFields": true
     },
     {
+      "name": "dev.langchain4j.model.anthropic.internal.api.AnthropicOutputFormat",
+      "allDeclaredConstructors": true,
+      "allPublicConstructors": true,
+      "allDeclaredMethods": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true,
+      "allPublicFields": true
+    },
+    {
+      "name": "dev.langchain4j.model.anthropic.internal.api.AnthropicOutputFormatType",
+      "allDeclaredConstructors": true,
+      "allPublicConstructors": true,
+      "allDeclaredMethods": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true,
+      "allPublicFields": true
+    },
+    {
       "name": "dev.langchain4j.model.anthropic.internal.api.AnthropicPdfContent",
       "allDeclaredConstructors": true,
       "allPublicConstructors": true,
Index: langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicMapperTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicMapperTest.java b/langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicMapperTest.java
--- a/langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicMapperTest.java	(revision 8680b0b82bbc8e4be2616345527f3fa199063c4e)
+++ b/langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/AnthropicMapperTest.java	(revision 00b2b4c2bf0d4f0acb2997f8f96d7e033bf92915)
@@ -3,7 +3,7 @@
 import static dev.langchain4j.model.anthropic.internal.api.AnthropicRole.ASSISTANT;
 import static dev.langchain4j.model.anthropic.internal.api.AnthropicRole.USER;
 import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.retainKeys;
-import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.toAnthropicMap;
+import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.toAnthropicSchema;
 import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.toAnthropicMessages;
 import static dev.langchain4j.model.anthropic.internal.mapper.AnthropicMapper.toAnthropicTool;
 import static java.util.Arrays.asList;
@@ -266,7 +266,7 @@
                 .build();
 
         // when
-        Map<String, Object> map = toAnthropicMap(jsonSchemaElement);
+        Map<String, Object> map = toAnthropicSchema(jsonSchemaElement);
 
         // then
         assertThat(new ObjectMapper().writeValueAsString(map))
@@ -299,7 +299,7 @@
                 .build();
 
         // when
-        Map<String, Object> map = toAnthropicMap(bookRecord);
+        Map<String, Object> map = toAnthropicSchema(bookRecord);
 
         // then
         assertThat(new ObjectMapper().writeValueAsString(map))

```

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java [hunk 4]
```diff
@@ -381,4 +384,70 @@ public static AnthropicTool toAnthropicTool(AnthropicServerTool serverTool) {
                 .customParameters(customParameters)
                 .build();
     }
+
+    public static Map<String, Object> toAnthropicMap(JsonSchemaElement schemaElement) {
+        return toAnthropicMap(schemaElement, true);
+    }
+
+    private static Map<String, Object> toAnthropicMap(JsonSchemaElement schemaElement, boolean required) {
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/mapper/AnthropicMapper.java]: `required` is not used anywhere, is it intended?

**Comment by dliubarskyi** [comment]
`should_extract_pojo_with_recursion` and `should_extract_pojo_with_local_date_time_fields` are failing when run locally, could you please check?

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java [hunk 5]
```diff
@@ -28,9 +30,6 @@ private InternalAnthropicHelper() {}
 
     static void validate(ChatRequestParameters parameters) {
         List<String> unsupportedFeatures = new ArrayList<>();
-        if (parameters.responseFormat() != null && parameters.responseFormat().type() == JSON) {
-            unsupportedFeatures.add("JSON response format");
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java:32]: Shouldn't we still fail when `parameters.responseFormat().type() == JSON && parameters.responseFormat().jsonSchema() == null` ? AFAIU "json mode" is not supported by anthropic?

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                .build();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public String toString() {
+        return "AnthropicResponseFormat[" + "type" + type + ", jsonSchema" + schema + "]";
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/api/AnthropicOutputFormat.java]:
```suggestion
        return "AnthropicOutputFormat[" + "type" + type + ", jsonSchema" + schema + "]";
```

File: langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/InternalAnthropicHelperTest.java [hunk 7]
```diff
@@ -23,21 +23,6 @@ void validate_WithNoUnsupportedFeatures_ShouldNotThrowException() {
         assertDoesNotThrow(() -> validate(parameters));
     }
 
-    @Test
-    void validate_WithJsonResponseFormat_ShouldThrowException() {
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/InternalAnthropicHelperTest.java:27]: I guess this test should stay? Json schema is supported now, but json format is still not

File: langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/common/AnthropicChatModelIT.java [hunk 8]
```diff
@@ -65,21 +78,43 @@ protected Class<? extends ChatResponseMetadata> chatResponseMetadataType(ChatMod
...
+    protected void should_execute_a_tool_then_answer_respecting_JSON_response_format_with_schema(ChatModel model) {
+        // Anthropic structured outputs are a public beta and only supported by
+        // Claude Sonnet 4.5, Opus 4.1/4.5, and Haiku 4.5 when the
+        // 'structured-outputs-2025-11-13' beta header is enabled.
+        super.should_execute_a_tool_then_answer_respecting_JSON_response_format_with_schema(model);
+    }
+
     @Override
     protected boolean supportsJsonResponseFormat() {
         // Anthropic does not support response format yet
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/common/AnthropicChatModelIT.java]:
```suggestion
        // Anthropic does not support JSON response format without a schema yet
```

File: langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/common/AnthropicChatModelIT.java [hunk 9]
```diff
@@ -65,21 +78,43 @@ protected Class<? extends ChatResponseMetadata> chatResponseMetadataType(ChatMod
...
         return false;
     }
 
     @Override
-    protected boolean supportsJsonResponseFormatWithSchema() {
-        // Anthropic does not support response format yet
-        return false;
+    @Disabled("Anthropic does not reliably support schemaless JSON structured outputs.")
+    protected void should_respect_JSON_response_format(ChatModel model) {
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/common/AnthropicChatModelIT.java]: No need to override this test, as `supportsJsonResponseFormat()` returns `false`, so the test will not be executed

File: langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/common/AnthropicStreamingChatModelIT.java [hunk 10]
```diff
@@ -130,4 +142,39 @@ protected void verifyToolCallbacks(StreamingChatResponseHandler handler, InOrder
...
+            StreamingChatModel model) {
+        // Anthropic structured outputs are a public beta and only supported by
+        // Claude Sonnet 4.5, Opus 4.1/4.5, and Haiku 4.5 when the
+        // 'structured-outputs-2025-11-13' beta header is enabled.
+        super.should_execute_a_tool_then_answer_respecting_JSON_response_format_with_schema(model);
+    }
+
+    @Override
+    @Disabled("Anthropic does not reliably support schemaless JSON structured outputs.")
+    protected void should_respect_JSON_response_format(StreamingChatModel model) {
```
Comment (dliubarskyi) [inline langchain4j-anthropic/src/test/java/dev/langchain4j/model/anthropic/common/AnthropicStreamingChatModelIT.java]: same here

**Comment by dliubarskyi** [comment]
> In this case, by “ignoring unsupported features,” do you mean using the existing toMap() and overriding the failing test (for example, disabling it or adapting it so that enums are marked as required in the schema)?

I misunderstood your previous explanations, please ignore )

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java (hunk 5 above)
Comment (dliubarskyi) [reply langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/InternalAnthropicHelper.java:32]: Seems like we should fail, the test is also not happy:

image

**Comment by dliubarskyi** [comment]
Could you please check this test failure:

image

**Comment by diegoberriosr** [comment]
Hey @dliubarskyi! Just resolved all comments, thank you.

Regarding failing tests:

- `should_extract_pojo_with_local_date_time_fields`: This one was flaky, even when using the existing `toMap()` if failed every now and then. I ended up disabling it (not quite sure if it is the best thing to do though):

image
<div style="margin-top: 5px;">
<br> </br>

- `should_extract_pojo_with_recursion`: Should be disabled, Anthropic does still not support recursive schemas (didn't override `supportsRecursion` in ` AbstractAiServiceWithJsonSchemaIT` for that reason):
 
image

<br> </br>
- `should_fail_if_json_response_format_with_schema_is_not_supported`: Should be skipped now:

image

<br> </br>
Here's a screenshot of a local run with all the tests in `langchain4j-anthropic`:

image

**Review by dliubarskyi — APPROVED** [review]
@diegoberriosr great job, thank you!
